source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -18,7 +18,69 @@ namespace Extensions {
 namespace HttpFilters {
 namespace HeaderToMetadataFilter {
 
-Rule::Rule(const std::string& header, const ProtoRule& rule) : header_(header), rule_(rule) {
+// extract the value of the header
```

Comment:
nit: i think most of the other comments are full sentences (e.g.: upper case at the start, period at the end).